Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+        Returns
+        -------
+        TableBase
+            A table converted from the source domain to the target domain.
+        """
+        # if a series is passed, convert it to a frame
+        if isinstance(source_table, SeriesBase):
+            d = source_table.domain
+            # to pandas df first to avoid adding a column/row of weights automatically
+            source_table = cls(source_table.domain, pd.DataFrame(source_table).transpose())
+
+        new_cache = cls._conversion_cache is None
+        try:
+            if new_cache:
+                cls._conversion_cache = {}
+            else:
+                cached = cls._conversion_cache.get((id(target_domain), id(source_table)))
+                if cached is not None:
+                    return cached
+            if target_domain == source_table.domain:
+                # intentional casting to subclass
+                result = cls(data=source_table.iloc[row_indices]).copy()
+                # because we manually copy data, not the whole table
+                result._transfer_properties(source_table, transfer_domain=True)
+                return result
+
+            result = cls()
+            conversion = target_domain.get_conversion(source_table.domain)
+
+            for conversion, target_column in zip(chain(conversion.variables, conversion.metas),
+                                                 chain(target_domain.variables, target_domain.metas)):
+                if isinstance(conversion, Number):
+                    # mandatory copy
+                    result[target_column.name] = source_table[source_table.domain[conversion]].copy()
+                else:
+                    # when converting a single instance (row), this results in a single value
+                    # (and not e.g. a Series), so assigning to a table fails:
+                    # we ensure it's at least a 1D ndarray, and not a 0D ndarray
+                    result[target_column.name] = np.atleast_1d(conversion(source_table))
+            result.domain = target_domain
+
+            # if the new domain has 0 columns, the length of the table is also 0
+            # and we can't set the previous weights or index (which are len(source_table))
+            if len(result) != 0:
+                result.set_weights(source_table.weights)
+                result.index = source_table.index  # keep previous index
+            result = result.iloc[row_indices]
+
+            cls._conversion_cache[(id(target_domain), id(source_table))] = result
+
+            # transform any values we believe are null into actual null values
+            result.replace(to_replace=list(Variable.MISSING_VALUES), value=np.nan, inplace=True)
+            result._transform_discrete_into_categorical()
+            result._transform_timevariable_into_datetime()
+
+            return result
+        finally:
+            if new_cache:
+                cls._conversion_cache = None
```

Comment:
It seems to me this is `TableBase._conversion_cache = None ` from start, and then here we have:
```py
new_cache = cls._conversion_cache is None
try:
    if new_cache:
        cls._conversion_cache = {}
    ...
finally:
    if new_cache:
        cls._conversion_cache = None
```
To me it looks like this resets in every passing. But I certainly don't know what is is / does.